### PR TITLE
Jes struk patch 1

### PR DESCRIPTION
**Pull request by JesStruk**
# Description

Updated the README and also deleted the ip groups from the module to make them a parameter.
Ip groups now can be placed in the template file.

# Change overview (tick true):

- [ ] README updated
- [ ] Deleted IP Groups as resource
- [ ] Added variables to take ip groups as input

# Version information: 

- Previous Version: `1.0.0`

- Next Version (see above): `1.0.1`

# How Has This Been Tested?

- [ ] Terraform Apply

# Checklist:

- [ ] I have run tests and documented them above
- [ ] I have performed a self-review of my own code
- [ ] I have updated the documentation

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -11,8 +11,8 @@ resource "azurerm_firewall_policy_rule_collection_group" "this" {
     rule {
       name                  = "allow-alz-to-dc-inbound"
       protocols             = ["TCP", "UDP"]
-      source_ip_groups      = [azurerm_ip_group.ipg_aplication_lz.id]
-      destination_ip_groups = [azurerm_ip_group.ipg_azure_dc.id, azurerm_ip_group.ipg_onprem_dc.id]
+      source_ip_groups      = [var.ipg_aplication_lz_id]
+      destination_ip_groups = [var.ipg_azure_dc_id, var.ipg_onpremise_dc_id]
```
Comment (ghost) [inline main.tf]:
```suggestion
      destination_ip_groups = [var.ipg_azure_dc_id]
```
There aren't always on-premise DCs and even if there are, a connection should only be needed to the cloud DCs. Confirm with Maurice

File: main.tf [hunk 2]
```diff
@@ -71,7 +71,7 @@ resource "azurerm_firewall_policy_rule_collection_group" "this" {
     rule {
       name                  = "allow-update-management-outbound"
       protocols             = ["TCP"]
-      source_ip_groups      = [azurerm_ip_group.ipg_aplication_lz.id]
+      source_ip_groups      = [var.ipg_aplication_lz_id]
```
Comment (ghost) [inline main.tf]: Only application LZs? Our DCs need updates too!

File: main.tf [hunk 3]
```diff
@@ -58,7 +58,7 @@ resource "azurerm_firewall_policy_rule_collection_group" "this" {
     rule {
       name             = "allow-backup-monitoring-outbound"
       protocols        = ["TCP"]
-      source_ip_groups = [azurerm_ip_group.ipg_aplication_lz.id]
+      source_ip_groups = [var.ipg_aplication_lz_id]
```
Comment (ghost) [inline main.tf]: Only application? DCs?

File: main.tf [hunk 4]
```diff
@@ -42,7 +42,7 @@ resource "azurerm_firewall_policy_rule_collection_group" "this" {
     rule {
       name              = "allow-alz-to-kms-outbound"
       protocols         = ["TCP"]
-      source_ip_groups  = [azurerm_ip_group.ipg_aplication_lz.id]
+      source_ip_groups  = [var.ipg_aplication_lz_id]
```
Comment (ghost) [inline main.tf]: Only application? DCs?

File: variables.tf [hunk 5]
```diff
@@ -27,44 +27,23 @@ variable "default_location" {
   nullable    = false
 }
 
-variable "ip_address_azure_dc" {
-  type        = set(string)
+variable "ipg_azure_dc_id" {
+  type        = string
   description = "The ip addresses of the domain controller located in azure. As standard the alz should only located in azure."
```
Comment (ghost) [inline variables.tf]:
```suggestion
  description = "The IP addresses of the domain controllers located in azure. As standard the ALZ should only be located in Azure."
```
"The ALZ (Azure Landing Zone) should only be located in Azure". What does this sentence mean?

File: variables.tf [hunk 6]
```diff
@@ -27,44 +27,23 @@ variable "default_location" {
...
+variable "ipg_onpremise_dc_id" {
+  type        = string
   description = "If the customer still operates domain controller on premise, provide these in this variable."
-  validation {
-    condition = alltrue(
-      [for value in var.ip_address_onprem_dc : can(regex("^(\\d{1,3}[.]){3}(\\d{1,3}[/]\\d{1,3}){1}$", value))]
-    )
-    error_message = "The provided ip address does not match the syntax of ddd.ddd.ddd.ddd/ddd"
-  }
+  default     = ""
```
Comment (ghost) [inline variables.tf]:
```suggestion
  default     = null
```
Setting default null is possible, right? Currently this adds an empty string into a rule, but it coul be a null check instead

File: variables.tf [hunk 7]
```diff
@@ -27,44 +27,23 @@ variable "default_location" {
...
-    )
-    error_message = "The provided ip address does not match the syntax of ddd.ddd.ddd.ddd/ddd"
-  }
+variable "ipg_dnsprivateresolver_id" {
+  type        = string
+  description = "The ip address of the private dns resolver."
 }
 
-variable "ip_address_dnsprivateresolver" {
+variable "ipg_aplication_lz_id" {
```
Comment (ghost) [inline variables.tf]:
```suggestion
variable "ipg_application_lz_id" {
```

File: variables.tf [hunk 8]
```diff
@@ -27,44 +27,23 @@ variable "default_location" {
...
-  description = "The application landing zone are ip ranges of applications that need to be added to the firewall rule set."
-  validation {
-    condition = alltrue(
-      [for value in var.ip_address_alz : can(regex("^(\\d{1,3}[.]){3}(\\d{1,3}[/]\\d{1,3}){1}$", value))]
-    )
-    error_message = "The provided ip address does not match the syntax of ddd.ddd.ddd.ddd/ddd"
-  }
+variable "ipg_dnsprivateresolver_id" {
+  type        = string
+  description = "The ip address of the private dns resolver."
```
Comment (ghost) [inline variables.tf]:
```suggestion
  description = "The IP address of the private dns resolvers inbound endpoint."
```

File: variables.tf [hunk 9]
```diff
@@ -27,44 +27,23 @@ variable "default_location" {
...
 
-variable "ip_address_dnsprivateresolver" {
+variable "ipg_aplication_lz_id" {
   type        = string
-  description = "The ip address of the private dns resolver for the ip group."
-  validation {
-    condition     = can(regex("^(\\d{1,3}[.]){3}(\\d{1,3}[/]\\d{1,3}){1}$", var.ip_address_DNSPrivateResolver))
-    error_message = "The provided ip address does not match the syntax of ddd.ddd.ddd.ddd/ddd"
-  }
+  description = "The application landing zone are ip ranges of applications that need to be added to the firewall rule set."
```
Comment (ghost) [inline variables.tf]:
```suggestion
  description = "IP ranges for all application landing zones."
```

File: main.tf [hunk 10]
```diff
@@ -28,8 +28,8 @@ resource "azurerm_firewall_policy_rule_collection_group" "this" {
     rule {
       name                  = "allow-dnsresolver-to-dc-inbound"
```
Comment (ghost) [inline main.tf]: Above inbound is used as left to right (alz->dc). Here inbound is used left to right (dnsresolver<-dc). Either change name or inbound/outbound

File: README.md [hunk 11]
```diff
@@ -1,10 +1,10 @@
-# Module
+# Platform Firewall Rules
 [![GitHub tag](https://img.shields.io/github/tag/qbeyond/terraform-module-template.svg)](https://registry.terraform.io/modules/qbeyond/terraform-module-template/provider/latest)
 [![License](https://img.shields.io/github/license/qbeyond/terraform-module-template.svg)](https://github.com/qbeyond/terraform-module-template/blob/main/LICENSE)
 
 ----
 
-This is a template module. It just showcases how a module should look. This would be a short description of the module.
+This module creates a firewall rule collection group with standardized rules that can be used. The standard ist defined from q.beyond.
```
Comment (ghost) [inline README.md]: Please add that it is for platform rules only and explain what platform means

Comment (ghost) [inline README.md]:
```suggestion
This module creates a firewall rule collection group with standardized rules. The standard ist defined by the q.beyond AG.
```

File: main.tf (hunk 2 above)
Comment (JesStruk) [reply main.tf]: Yes, maybe. :-D

File: main.tf (hunk 3 above)
Comment (JesStruk) [reply main.tf]: Yes

File: main.tf (hunk 4 above)
Comment (JesStruk) [reply main.tf]: Yes

File: main.tf (hunk 1 above)
Comment (JesStruk) [reply main.tf]: Please check, made changes to check if on prem dc is added and if not, use list without.

File: main.tf (hunk 2 above)
Comment (JesStruk) [reply main.tf]: Please check, made changes to check if on prem dc is added and if not, use list without.

File: main.tf (hunk 3 above)
Comment (JesStruk) [reply main.tf]: Please check, made changes to check if on prem dc is added and if not, use list without.

File: main.tf (hunk 4 above)
Comment (JesStruk) [reply main.tf]: Please check, made changes to check if on prem dc is added and if not, use list without.

File: variables.tf (hunk 5 above)
Comment (JesStruk) [reply variables.tf]: Deleted, was a stupid sentence xD

File: variables.tf (hunk 6 above)
Comment (JesStruk) [reply variables.tf]: Done

**Review by ghost — APPROVED** [review]

**Review by TheodorWulff — APPROVED** [review]

**Pull request merged**